tools/hsmtool.c

```diff
@@ -368,6 +371,145 @@ static int guess_to_remote(const char *address, struct node_id *node_id,
 	return 1;
 }
 
+static void get_words(struct words **words) {
+	printf("Select your language:\n");
+	printf("  0) English (en)\n");
+	printf("  1) Spanish (es)\n");
+	printf("  2) French (fr)\n");
+	printf("  3) Italian (it)\n");
+	printf("  4) Japanese (jp)\n");
+	printf("  5) Chinese simplified (zhs)\n");
+	printf("  6) Chinese traditional (zht)\n");
+	printf("Select [0-6]: ");
+
+	char *selection=NULL;
+	size_t size = 0;
+	size_t characters = getline(&selection, &size, stdin);
+	if (characters < 0)
+		errx(ERROR_USAGE, "Could not read line from stdin.");
+	int selected = atoi(selection);
+
+	char *lang = malloc(4);
+	switch (selected) {
+		case 0:
+			strcpy(lang, "en");
+			break;
+		case 1:
+			strcpy(lang, "es");
+			break;
+		case 2:
+			strcpy(lang, "fr");
+			break;
+		case 3:
+			strcpy(lang, "it");
+			break;
+		case 4:
+			strcpy(lang, "jp");
+			break;
+		case 5:
+			strcpy(lang, "zhs");
+			break;
+		case 6:
+			strcpy(lang, "zht");
+			break;
+		default:
+			errx(ERROR_USAGE, "Invalid language selection, select one from the list [0-6].");
```

Comment:
You could use `strtol` or similar

Reply:
>  This cannot be reached, `atoi` will give you `0` if throw it garbage

This is true, and I did not know it! Maybe the safest thing to do then is to take the first byte from the user's input and subtract '0' instead of doing any str to number conversion.

Reply:
Hmm i'd say the safest thing here to go for `strtol`, which errors on --well-- error. From the manpage:
> The program shown below demonstrates the use of strtol().  The first command-line argument specifies a string from which strtol() should parse a number.  The second (optional) argument  specifies  the
       base to be used for the conversion.  (This argument is converted to numeric form using atoi(3), a function that performs no error checking and has a simpler interface than strtol().)  Some examples of
       the results produced by this program are the following:

           $ ./a.out 123
           strtol() returned 123
           $ ./a.out '    123'
           strtol() returned 123
           $ ./a.out 123abc
           strtol() returned 123
           Further characters after number: abc
           $ ./a.out 123abc 55
           strtol: Invalid argument
           $ ./a.out ''
           No digits were found
           $ ./a.out 4000000000
           strtol: Numerical result out of range